contrib/babelfishpg_tds/src/backend/tds/tdstimestamp.c

```diff
@@ -423,7 +423,9 @@ TdsGetDayTimeFromTimestamp(Timestamp value, uint32 *numDays, uint64 *numSec,
 			((double)fsec/ 1000000));
 	while (scale--)
 		res *= 10;
-
+	/* Round res to the nearest integer */
```

Comment:
There is an issue that was also covered in the JIRA where occasionally some dates would not get displayed correctly. I determined that this was due to errors in floating point arithmetic that would for example, change "0.000012" to "0.000011999998" internally. When this gets converted back to an int64 for display, it gets truncated to "0.000011". This is why I add 0.5 prior to casting to an int; this has the effect of having the double rounded to the nearest integer instead of just getting floored down all of the time.